tests/smoketest/smoketest.sh

```diff
@@ -59,7 +68,7 @@ for NAME in "${CLOUDNAMES[@]}"; do
 done
 
 echo "*** [INFO] Triggering an alertmanager notification..."
-PROMETHEUS_K8S_TOKEN=$(oc serviceaccounts get-token prometheus-k8s)
+PROMETHEUS_K8S_TOKEN=$(oc create token prometheus-k8s)
```

Comment:
I don't have a 4.11 cluster handy to test this, but it looks suspicious to me. Possibly it's an inconsistency coming from openshift? Usually the create command creates a new resource in the cluster and return a string like 
`token.some411api.openshift.io "prometheus-k8s" created`, but here you are expecting it to return the actual token? 

I read the 4.11 release notes about token changes[1], but they don't seem to describe this workflow. I did find this CLI reference[2] which makes it sound like it might do what you show it doing.

I believe you, but needed to look this up since it seems weird.

[1] https://docs.openshift.com/container-platform/4.11/release_notes/ocp-4-11-release-notes.html#ocp-4-11-notable-technical-changes 
[2] https://docs.openshift.com/container-platform/4.11/cli_reference/openshift_cli/developer-cli-commands.html#oc-create-token

Reply:
So the original comment was doing a `get-token` which I also assumed was returning the token string for assignment to PROMETHEUS_K8S_TOKEN. I wasn't able to use this with my `oc` client of version 4.7 when doing the testing as it was failing. I ran the command locally, and it wouldn't return anything for me. I did some digging online for what replaced the deprecated command we were using, and it seemed to imply I should use the new `create` command.

I also found this weird. It seems the entire `serviceaccounts` subcommand goes away in favour of `oc create token`. I was checking the Secrets and such to determine if it was creating new objects, but that didn't seem to be the case.

Looking at what is happening in the smoketest, I believe returning a token string is appropriate, as the string is passed as a bearer token to the `curl` command to trigger an alert being fired (and eventually consumed by SNMP webhook service) via the alertmanager API.

If I run `oc serviceaccounts` I get this output:

```
Command "serviceaccounts" is deprecated, and will be removed in the future version. Use oc create token instead.
Manage service accounts in your project

 Service accounts allow system components to access the API.

Aliases:
serviceaccounts, sa

Usage:
  oc serviceaccounts [flags] [options]

Use "oc <command> --help" for more information about a given command.
Use "oc options" for a list of global command-line options (applies to all commands).
```

So I looked at that output, which provides some of this:

```
oc create token --help
Request a service account token.

Examples:
  # Request a token to authenticate to the kube-apiserver as the service account "myapp" in the current namespace
  oc create token myapp
```

There are other options for how long the token is valid for and which object the token is bound to.

Reply:
Also, I ran the command a few times and then ran `oc get secrets | wc -l` and the value never increases. Are there other object types I should verify against?

Reply:
:+1: If it works it works. Seems to do what it says, just looks weird.